static/js/draw_charts.tsx

```diff
@@ -0,0 +1,181 @@
+  chartElem: string;
+  placeIds: string[];
+  statVarsAndMeasuredProps: any[];
+  perCapita: boolean;
+}
+
+class ChartRegion extends Component<ChartRegionPropsType, {}> {
+  measuredPropGroup: { string?: string[] };
+  measuredProps: string[];
+  chartIds: string[];
+  width: number;
+  height: number;
+
+  constructor(props) {
+    super(props);
+
+    // get mprop-statVar dict: { mprop: [statVar1, statVar2, ...]}
+    this.measuredPropGroup = {};
+    this.measuredProps = [];
+    for (let statVarAndMeasuredProp of this.props.statVarsAndMeasuredProps) {
+      let mprop = statVarAndMeasuredProp[1];
+      if (mprop in this.measuredPropGroup) {
+        this.measuredPropGroup[mprop].push(statVarAndMeasuredProp[0]);
+      } else {
+        this.measuredProps.push(mprop);
+        this.measuredPropGroup[mprop] = [statVarAndMeasuredProp[0]];
+      }
+    }
+
+    // get width and height for chart
+    const obsElem = document.getElementById(this.props.chartElem);
+    this.width = Math.min(obsElem.offsetWidth - 20, MAX_CHART_WIDTH);
+    this.height = Math.min(Math.round(this.width * 0.5), MAX_CHART_HEIGHT);
+
+    let length = Object.keys(this.measuredPropGroup).length;
+    this.chartIds = [];
+    for (let i = 0; i < length; i++) {
+      this.chartIds.push(randDomId());
+    }
+
+    // Empty state in the beginning, only dataGroupDict will be in the state
+    this.state = {};
+  }
+
+  componentDidMount() {
+    // Set up states
+    let promises: Promise<StatsData>[] = [];
+    let mprops = [];
+    for (let mprop in this.measuredPropGroup) {
+      mprops.push(mprop);
+      let statsVarsArray = this.measuredPropGroup[mprop];
+      // Make an array of Promises
+      promises.push(
+        fetchStatsData(
+          this.props.placeIds,
+          statsVarsArray,
+          this.props.perCapita,
+          1
+        )
+      );
+    }
+
+    Promise.all(promises).then((statDatas) => {
+      let state = {};
+      let params = {};
+      for (let i = 0; i < statDatas.length; i++) {
+        // generate dict {geoId: DataGroup}.
+        let dataGroupsDict = {};
+        for (let geo of statDatas[i].places) {
+          dataGroupsDict[geo] = statDatas[i].getStatsVarGroupWithTime(geo);
+        }
+        state[mprops[i]] = dataGroupsDict;
+        params[mprops[i]] = computePlotParams(dataGroupsDict);
+      }
+      this.setState({ data: state, params: params });
+    });
+  }
+
+  componentDidUpdate() {
+    this.updateChart();
+  }
+
+  updateChart() {
+    let index = 0;
+    for (let mprop in this.state["data"]) {
+      let dataGroupsDict = this.state["data"][mprop];
+      let elemId = this.chartIds[index];
+      drawGroupLineChart(
+        elemId,
+        this.width,
+        this.height,
+        dataGroupsDict,
+        this.state["params"][mprop]
+      );
+      index++;
+    }
+  }
+
+  deleteStatVarChip(statVar: string) {
+    // TODO: add function to delete statvarchip.
+    return;
+  }
+
+  render() {
```

Comment:
can you put render() the first function after constructor()

Reply:
Done